Mu2eKinKal/src/LoopHelixFit_module.cc

```diff
@@ -318,12 +339,43 @@ namespace mu2e {
       kkbf_ = std::move(std::make_unique<KKBField>(*bfmgr,*det));
     }
     if(print_ > 0) kkbf_->print(std::cout);
+
+    // Print the fit direction configuration
+    if(print_ > -1) printf("[LoopHelixFit::%s::%s] Fit dz/dt direction = %.0f, PDG = %i, use helix slope = %o with threshold %.1f, use helix slope cut = %o with theshold %.1f\n",
+                           __func__, moduleDescription().moduleLabel().c_str(), fdir_.dzdt(), (int) fpart_,
+                           useHelixSlope_, (useHelixSlope_) ? slopeSigThreshold_ : -1.f,
+                           useSlopeSigCut_, (useSlopeSigCut_) ? slopeSigCut_ : -999.f);
   }
 
-  void LoopHelixFit::produce(art::Event& event ) {
-    // empty collections
-    static MEASCOL nohits; // empty
-    static EXINGCOL noexings; // empty
+  HelixRecoDir::PropDir LoopHelixFit::chooseHelixDir(HelixSeed const& hseed) const {
+    if(useHelixSlope_) return hseed.recoDir().predictDirection(slopeSigThreshold_);
+    if(fdir_.fitDirection() == TrkFitDirection::FitDirection::upstream  ) return HelixRecoDir::PropDir::upstream;
+    if(fdir_.fitDirection() == TrkFitDirection::FitDirection::downstream) return HelixRecoDir::PropDir::downstream;
+    return HelixRecoDir::PropDir::ambiguous;
+  }
+
+  bool LoopHelixFit::compare_tracks(KKTRK const& ktrk_1, KKTRK const& ktrk_2) const {
+    // return true to select the first track, false to select the second
+
+    const bool goodfit_1(goodFit(ktrk_1)), goodfit_2(goodFit(ktrk_2));
+    if(print_>1) printf(" [LoopHelixFit::%s]: status 1 = %o, status 2 = %o\n", __func__, goodfit_1, goodfit_2);
+
+    if(!goodfit_2) return true; //if the second is bad or if both are bad, default to the first
```

Comment:
If the function returns on this statement the first fit could still be bad.
Returning a more complicated return value than 'bool' would address this, by having 3 states (1 is better, 2 is better, neither are good)

Reply:
If both fits are bad, this is caught outside this comparison in the requirement that the accepted track is a good fit. In the case all fits are meant to be saved, this defaults to saving the first track given to the comparison. I updated to return 1 if the first is better, 0 if the second is better, and -1 if both are bad fits, still defaulting to the first track if both are bad.